src/domain/hooks/hooks.repository.ts

```diff
@@ -73,12 +77,28 @@ export class HooksRepositoryWithNotifications implements IHooksRepository {
   }
 
   async onEvent(event: Event): Promise<unknown> {
-    return Promise.allSettled([
-      this.onEventClearCache(event),
-      this.eventNotificationsHelper.onEventEnqueueNotifications(event),
-    ]).finally(() => {
-      this.onEventLog(event);
-    });
+    const isSupportedChainId = await this.isSupportedChainMemo(event.chainId);
+    if (isSupportedChainId) {
+      return Promise.allSettled([
+        this.onEventClearCache(event),
+        this.eventNotificationsHelper.onEventEnqueueNotifications(event),
+      ]).finally(() => {
+        this.onEventLog(event);
+      });
+    } else {
+      this.loggingService.warn({
+        type: 'unsupported_event_chain',
+        chainId: event.chainId,
+        eventType: event.type,
+      });
+    }
+  }
+
+  private async isSupportedChain(chainId: string): Promise<boolean> {
```

Comment:
Nit: I think this should "live" in the chains repository and we then expose a `clearSupportedChain` method for consistency. What do you think?

Reply:
Moved in [789cee5](https://github.com/safe-global/safe-client-gateway/pull/1920/commits/789cee533294c8d1e6717e74004882002c737b01)

Reply:
I didn't move the memoized function to the `ChainsRepository` because I think it could add more complexity (other classes might not need to call a memoized function I think). So I've left the clearing of the memoized function cache responsibility in the `HooksRepository` for now. If memoizing makes sense in other high-load places we could think on memoizing inside the `ChainsRepository` (or even on other repositories), but I tend to think it is out of the scope of the PR, wdyt?